File: python/GPU.py
from Komponen import Komponen

class GPU(Komponen):
    def __init__(self, nama_seri="", jumlah_memori=0, tipe_memori="", nama="", merk="", tahun_rilis=0, harga=0):
        super().__init__(nama, merk, tahun_rilis, harga)
        self.nama_seri = nama_seri
        self.jumlah_memori = jumlah_memori
        self.tipe_memori = tipe_memori
    
    def setNamaSeri(self, nama_seri):
        self.nama_seri = nama_seri
    
    def setJumlahMemori(self, jumlah_memori):
        self.jumlah_memori = jumlah_memori
    
    def setTipeMemori(self, tipe_memori):
        self.tipe_memori = tipe_memori
    
    def getNamaSeri(self):
        return self.nama_seri
    
    def getJumlahMemori(self):
        return self.jumlah_memori
    
    def getTipeMemori(self):
        return self.tipe_memori

File: cpp/GPU.cpp
#pragma once
#include "Komponen.cpp"
#include <iostream>
#include <string>
using namespace std;

class GPU : public Komponen
{
private:
    string nama_seri;
    int jumlah_memori;
    string tipe_memori;

public:
    GPU() {}

    GPU(string nama_seri, int jumlah_memori, string tipe_memori, string nama, string merk, int tahun_rilis, int harga) : Komponen(nama, merk, tahun_rilis, harga)
    {
        this->nama_seri = nama_seri;
        this->jumlah_memori = jumlah_memori;
        this->tipe_memori = tipe_memori;
    }

    void setNamaSeri(string nama_seri)
    {
        this->nama_seri = nama_seri;
    }

    void setJumlahMemori(int jumlah_memori)
    {
        this->jumlah_memori = jumlah_memori;
    }

    void setTipeMemori(string tipe_memori)
    {
        this->tipe_memori = tipe_memori;
    }

    string getNamaSeri()
    {
        return nama_seri;
    }

    int getJumlahMemori()
    {
        return jumlah_memori;
    }

    string getTipeMemori()
    {
        return tipe_memori;
    }

    ~GPU() {}
};